### PR TITLE
Receive timeout

### PR DESCRIPTION
**Pull request by gdooper**
- Adds a new config `ReceiveTimeout` that lets you control the maximum amount of time a PartitionReceiver will attempt to get events from an eventhub.
- Updates the version to 0.6.1

---
This change is [Reviewable](https://reviewable.io/reviews/azure/toketi-kafka-connect-iothub/13)

### CODE REVIEW

File: README_Source.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ Here are the configurable values -
 | IotHub.StartTime | String | No | (Unused if not supplied) | The time from which to start retrieving messages from IoT Hub. The value should be in UTC and in the format yyyy-mm-ddThh:mm:ssZ. This setting is mutually exclusive with IotHub.Offsets. |
 | IotHub.Offsets | String | No | (Unused if not supplied) | The offsets for each IoT Hub partition from which to start retrieving messages from IoTHub, as a comma separated string. For example, for 4 partitions, the value would be something like "abc,lmn,pqr,xyz". This setting is mutually exclusive with IotHub.StartTime. |
 | BatchSize | Int | No | 100 | The size of each batch for retrieving entries from IoT Hub. |
+| RequestTimeout | Int | No | 60 | The max duration in seconds to spend receiving entries from IoT Hub. |
```
Comment (varunpuranik) [inline README_Source.md]: ReceiveTimeout

File: README_Source.md [hunk 2]
```diff
@@ -49,6 +50,7 @@ IotHub.Partitions=4
 IotHub.StartTime=2016-11-28T00:00:00Z
 IotHub.Offsets=
 BatchSize=100
+RequestTimeout=60
```
Comment (varunpuranik) [inline README_Source.md]: ReceiveTimeout

File: connect-iothub-source.properties [hunk 3]
```diff
@@ -56,3 +56,6 @@ IotHub.Offsets=PLACEHOLDER
 
 # The size of each batch for retrieving entries from IoTHub. The max supported value is 999.
 BatchSize=100
+
+# The max duration in seconds to spend receiving entries from IoTHub. The default is 60.
+ReceiveTimeout=60
```
Comment (varunpuranik) [inline connect-iothub-source.properties]: can we make this ReceiveTimeout

File: README_Source.md (hunk 1 above)
Comment (varunpuranik) [reply README_Source.md]: Also, the text is a bit confusing (it feels like the connector will stop after the timeout period). Can we reword it to something like - "Max duration to wait when retrieving entities from IoT Hub".

**Review by varunpuranik — CHANGES_REQUESTED** [review]
While the PR looks good to me, I would like to understand why you are having to set the ReceiveTimeout. Can you please comment on that?

**Comment by gdooper** [comment]
@varunpuranik In our message pipelines, we have some eventhubs that have messages trickle into the eventhub. Say 1 new message every 10 seconds. We want to keep the latencies in the pipeline as low as possible. Waiting up to 60 seconds for the batch of new events to fill to a count of `BatchSize` is not acceptable for us. Exposing this `ReceiveTimeout` as a config option allows us to reduce the latency without having to reduce the `BatchSize` parameter.

File: README_Source.md (hunk 1 above)
Comment (gdooper) [reply README_Source.md]: OK, I can fix this.

File: README_Source.md (hunk 2 above)
Comment (gdooper) [reply README_Source.md]: Oops. Will fix.

File: README_Source.md (hunk 1 above)
Comment (gdooper) [reply README_Source.md]: @varunpuranik For a description, how about `The max duration in seconds to wait for a full batch when retrieving messages from IoT Hub.`

Comment (varunpuranik) [reply README_Source.md]: Yes, that works..

**Comment by varunpuranik** [comment]
I see.. makes sense. Thanks for the explanation.

File: connect-iothub-source.properties (hunk 3 above)
Comment (gdooper) [reply connect-iothub-source.properties]: no change necessary here, right?

**Comment by gdooper** [comment]
@varunpuranik I've updated the PR. Can you review again?

**Review by varunpuranik — APPROVED** [review]
